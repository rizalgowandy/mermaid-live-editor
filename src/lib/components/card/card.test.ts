import { cleanup, render } from '@testing-library/svelte';
import { describe, expect, it, afterEach } from 'vitest';
import Card from './card.svelte';

describe('card.svelte', () => {
	// TODO: @testing-library/svelte claims to add this automatically but it doesn't work without explicit afterEach
	afterEach(() => cleanup());

	it('mounts', () => {
		const { container } = render(Card, {
			title: 'TabTest',
			tabs: [
				{ id: 't1', title: 'title1' },
				{ id: 't2', title: 'title2' }
			]
		});
		expect(container).toBeTruthy();
		expect(container).toHaveTextContent('TabTest');
		expect(container).toHaveTextContent('title1');
		expect(container).toHaveTextContent('title2');
		expect(container).not.toHaveTextContent('title3');
		expect(container.innerHTML).toMatchSnapshot();
	});
});
