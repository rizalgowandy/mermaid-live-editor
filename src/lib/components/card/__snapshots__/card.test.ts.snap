// Vitest Snapshot v1

exports[`card.svelte > mounts 1`] = `"<div><div class=\\"card rounded overflow-hidden m-2 flex-grow flex flex-col shadow-2xl\\"><div class=\\"bg-primary p-2 pb-0 flex-none cursor-pointer\\"><div class=\\"flex justify-between\\"><div class=\\"flex cursor-default s-_wx1E_JHsCoF\\"><span class=\\"mr-2 font-semibold s-_wx1E_JHsCoF\\"><i class=\\"fas fa-chevron-right icon s-_wx1E_JHsCoF isOpen\\"></i> TabTest</span> <ul class=\\"tabs s-_wx1E_JHsCoF\\"><div class=\\"tab tab-lifted tab-active s-_wx1E_JHsCoF\\"><i class=\\"mr-1 undefined s-_wx1E_JHsCoF\\"></i> title1 </div><div class=\\"tab tab-lifted text-primary-content s-_wx1E_JHsCoF\\"><i class=\\"mr-1 undefined s-_wx1E_JHsCoF\\"></i> title2 </div></ul></div><!--<Tabs>--> <div class=\\"flex gap-x-4 items-center -mt-2\\"></div></div></div> <div class=\\"card-body p-0 flex-grow overflow-auto text-base-content\\"></div></div><!--<Card>--></div>"`;
